lib/commands.js

```diff
@@ -358,13 +359,26 @@ var operations = module.exports.operations = {
     }
 
     var diff = compare(context.oldParameters, context.newParameters);
+
     if (!diff) {
       context.overrides.skipConfirmParameters = true;
       return context.next();
     }
 
+    if (context.overrides.diffs && context.overrides.diffs.parameters) {
+      var preapproved = context.overrides.diffs.parameters.filter(function(previous) {
+        return previous === diff;
```

Comment:
Is order a concern here - if parameters are in a different order in overrides.diff, but the values are the same? Does `===` catch this?

Reply:
👍 It would also be super helpful to read tests that help explain ^ better.

Reply:
I'm trying to consider a situation in which this would arise. 

If you jostle the order of the Parameters in your template, but don't change the parameter values, that will show up as a template change later on. 

If you have a stack in 2 regions, and they both have Parameters in a certain order, then you deploy a template change that jostles parameter order, you will be prompted to confirm parameters for the first region, but then the diff string will be identical on the 2nd region, and you won't be prompted.

If the 2 templates have different starting points, i.e. they were from 2 versions of a template and have parameters out-of-order with respect to each other, a deploy that brings them both to a new, 3rd template state will require you to confirm parameters twice.

I'm pretty bullish on being strict about string matching here -- you only get to skip confirmation steps if you are about to be prompted to confirm _exactly the same string_ that you've already been prompted with and confirmed.

I'm not sure if I managed to answer this concern here -- let me know!

Reply:
I thought that property order on Javascript objects is not guaranteed. So my concern was not that a different order had been dictated by a user on purpose (e.g. by a template change), but instead that the object would present properties in a different order without user input, simply because the property order is not guaranteed (e.g. unlike an array).